guard/lib/guard/front_repo/user.ex

```diff
@@ -229,4 +239,59 @@ defmodule Guard.FrontRepo.User do
 
     invalid_token_string or exists_by_token
   end
+
+  @doc """
+  Returns true if the user is a service account.
+  """
+  def service_account?(%__MODULE__{creation_source: :service_account}), do: true
+  def service_account?(_user), do: false
+
+  @doc """
+  Changeset for creating a service account user.
+  This validates the specific requirements for service account users.
+  """
+  def service_account_changeset(user, params) do
+    user
+    |> cast(params, [
+      :email,
+      :name,
+      :company,
+      :authentication_token,
+      :salt,
+      :creation_source,
+      :single_org_user,
+      :org_id,
+      :idempotency_token
+    ])
+    |> validate_required([:email, :name, :creation_source, :org_id])
+    |> validate_length(:name, max: 255, message: "Name cannot exceed 255 characters")
+    |> validate_inclusion(:creation_source, [:service_account])
+    |> put_change(:single_org_user, true)
+    |> validate_format(:email, ~r/^[\w\-\.]+@sa\.[\w\-\.]+\.semaphoreci\.com$/i,
+      message:
+        "Service account email must follow the format: name@sa.organization.semaphoreci.com"
+    )
+    |> unique_constraint(:email, name: :index_users_on_email)
+    |> unique_constraint(:authentication_token, name: :index_users_on_authentication_token)
+    |> unique_constraint(:idempotency_token, name: "users_idempotency_token_index")
+  end
+
+  @doc """
+  Generates a synthetic email for a service account.
+  """
+  def generate_service_account_email(service_account_name, organization_name) do
+    # Sanitize names to ensure valid email format
+    sanitized_sa_name = sanitize_email_part(service_account_name)
+    sanitized_org_name = sanitize_email_part(organization_name)
+
+    "#{sanitized_sa_name}@sa.#{sanitized_org_name}.semaphoreci.com"
```

Comment:
@DamjanBecirovic thoughts on this?

Reply:
The format looks good, but consider changing the `sa` to either `service-accounts` or `service_accounts`, as the `sa` abbreviation is not very common and may not be recognizable.

Reply:
I also removed hardcoded "semaphoreci.com" and used `BASE_DOMAIN` env var.